src/org/mozilla/javascript/NativeObject.java

```diff
@@ -544,29 +544,29 @@ public Object execIdCall(IdFunctionObject f, Context cx, Scriptable scope,
             if (args.length < 1) {
               throw ScriptRuntime.typeError1("msg.incompat.call", "assign");
             }
-            Scriptable t = ScriptRuntime.toObject(cx, thisObj, args[0]);
+            Scriptable targetObj = ScriptRuntime.toObject(cx, thisObj, args[0]);
             for (int i = 1; i < args.length; i++) {
               if ((args[i] == null) || Undefined.instance.equals(args[i])) {
```

Comment:
In other parts of the code I think that we started moving to using "Undefined.isUndefined()" for this test to ensure that the check works in all situations.